parquet-variant/src/main/java/org/apache/parquet/variant/VariantSizeLimitException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.parquet.variant;
+
+/**
+ * An exception indicating that the metadata or data size of the Variant exceeds the
+ * configured size limit.
+ */
+public class VariantSizeLimitException extends RuntimeException {
+  public VariantSizeLimitException(long sizeLimitBytes, long estimatedSizeBytes) {
+    super(String.format(
```

Comment:
Why is there a size limit imposed by Parquet? We don't do this anywhere else and it isn't in the spec.

Reply:
Sure, but this can be done outside of Parquet. Engines can reject large objects if they choose.